eclipse-collections/src/main/java/org/eclipse/collections/impl/lazy/FlatCollectIterable.java

```diff
@@ -78,6 +79,24 @@ public V detect(Predicate<? super V> predicate)
     }
 
     @Override
+    public Optional<V> detectOptional(Predicate<? super V> predicate)
+    {
+        V[] result = (V[]) new Object[1];
+        Iterate.detectOptional(this.adapted, each -> {
```

Comment:
The inner call to detectOptional() here, and detect() in the previous method, is just used to short-circuit. The result is not stored in a variable. I think it would be more clear to use anySatisfy() in both methods, in a way similar to the inner-most call to anySatisfy(). The way it is written will work, but it makes it seem like the Optional returned by detectOptional() is significant.